apps/backend/src/modules/gradtrak/typedefs/gradtrak.ts

```diff
@@ -0,0 +1,235 @@
+    """
+    secondary_section_IDs: [String!]
+    """
+    Requirements satisfied by class
+    """
+    uni_reqs: [String!]
+    college_reqs: [String!]
+  }
+
+  type CustomEvent {
+    start_time: String!
+    end_time: String!
+    title: String
+    location: String
+    description: String
+    days_of_week: String
+    uni_reqs: [String!]
+    college_reqs: [String!]
+  }
+
+  input MajorReqInput {
+    name: String!
+    major: String!
+    num_courses_required: Int!
+    satisfying_course_ids: [String!]!
+    is_minor: Boolean! 
+  }
+
+  input GqlTermInput {
+    year: Int!
+    planTerm: String!
+  }
+
+  input CustomEventInput {
+    start_time: String!
+    end_time: String!
+    title: String
+    location: String
+    description: String
+    days_of_week: String
+    uni_reqs: [Uni_Reqs!]
+    college_reqs: [College_Reqs!]
+  }
+
+  input SelectedCourseInput {
+    class_ID: String!
+    primary_section_ID: String
+    secondary_section_IDs: [String!]
+    uni_reqs: [Uni_Reqs!]
+    college_reqs: [College_Reqs!]
+  }
+
+  input PlanTermInput {
+    name: String
+    term: GqlTermInput
+    courses: [SelectedCourseInput!]
+    custom_events: [CustomEventInput!]
+  }
+
+  type Query {
+    """
+    Takes in user's email and returns their entire gradtrak
+    """
+    gradtrakByUser: Gradtrak @auth
+    
+    """
+    Takes in an ID and returns the planTerm for that term.
+    """
+    planTermByID(id: ID!): PlanTerm
+  }
+
+  type Mutation {
+    """
+    Takes in user's email, creates a new Gradtrak record in the database, and returns the Gradtrak
+    """
+    createNewGradtrak: Gradtrak @auth
+
+    """
+    Takes in a new college, switches the user's college requirements
+    """
+    changeGradtrakCollege(college: Colleges!): Gradtrak @auth
+
+    """
+    Takes in PlanTerm fields, creates a new PlanTerm record in the database, and returns the PlanTerm.
+    """
+    createNewPlanTerm(planTerm: PlanTermInput!): PlanTerm @auth
+
+    """
+    Takes in a PlanTerm's ObjectID, deletes the PlanTerm with that ID, and returns the ID.
+    """
+    removePlanTermByID(id: ID!): ID @auth
+    
+    """
+    Takes in planTerm fields, find the planTerm record in the database corresponding to the provided term, 
+    updates the record, and returns the updated planTerm
+    """
+    editPlanTerm(id: ID!, planTerm: PlanTermInput!): PlanTerm @auth
+    
+    """
+    For the planTerm specified by the term, modifies the courses field, and returns the updated 
+    planTerm.
+    """
+    setSelectedClasses(id: ID!, courses: [SelectedCourseInput!]!, custom_events: [CustomEventInput!]!): PlanTerm @auth
+
+    """
+    Edits the list of major requirements
+    """
+    editMajorRequirements(major_reqs: [MajorReqInput!]!): Gradtrak @auth
+
+    """
+    Deletes gradtrak, for testing purposes
+    """
+    deleteGradtrak: String @auth
```

Comment:
I would suggest naming this `Term` and not `Gradtrak` like I was talking about before.